scripts/system/create/entityProperties/html/js/materialAssistant.js

```diff
@@ -0,0 +1,890 @@
+        maRoughnessMap.disabled = false;
+    } 
+
+    if (materialDefinition.roughness !== undefined) {
+        maMaterialData.roughness = materialDefinition.roughness;
+    } else {
+        maMaterialData.roughness = DEFAULT_ROUGHNESS;
+    }
+    maRoughness.value = maMaterialData.roughness;
+    maRoughnessSlider.value = Math.floor(maMaterialData.roughness * 1000);
+    
+    if (materialDefinition.roughnessMap !== undefined) {
+        maMaterialData.roughnessMap = materialDefinition.roughnessMap;
+    } else {
+        maMaterialData.roughnessMap = "";
+    }
+    maRoughnessMap.value = maMaterialData.roughnessMap;
+
+    //NORMAL MAP
+    if (materialDefinition.defaultFallthrough === true && materialDefinition.normalMap === undefined) {
+        maMaterialData.normalMapIsActive = false;
+        maNormalMapIsActive.className = "uiMaterialAssistant-inactive";
+        maNormalMap.disabled = true;
+    } else {
+        maMaterialData.normalMapIsActive = true;
+        maNormalMapIsActive.className = "uiMaterialAssistant-active";
+        maNormalMap.disabled = false;
+    }
+
+    if (materialDefinition.normalMap !== undefined) {
+        maMaterialData.normalMap = materialDefinition.normalMap;
+    } else {
+        maMaterialData.normalMap = "";
+    }
+    maNormalMap.value = maMaterialData.normalMap;
+
+    //OPACITY
+    if (materialDefinition.defaultFallthrough === true 
+            && materialDefinition.opacity === undefined 
+            && materialDefinition.opacityMapMode === undefined) {
+        maMaterialData.opacityIsActive = false;
+        maOpacityIsActive.className = "uiMaterialAssistant-inactive";
+        maOpacity.disabled = true;
+        maOpacitySlider.disabled = true;
+        maOpacityMapModeDont.disabled = true;
+        maOpacityMapModeMask.disabled = true;
+        maOpacityMapModeBlend.disabled = true;
+        maOpacityCutoff.disabled = true;
+        maOpacityCutoffSlider.disabled = true;
+    } else {
+        maMaterialData.opacityIsActive = true;
+        maOpacityIsActive.className = "uiMaterialAssistant-active";
+        maOpacity.disabled = false;
+        maOpacitySlider.disabled = false;
+        maOpacityMapModeDont.disabled = false;
+        maOpacityMapModeMask.disabled = false;
+        maOpacityMapModeBlend.disabled = false;
+        maOpacityCutoff.disabled = false;
+        maOpacityCutoffSlider.disabled = false;
+    }
+    
+    if (materialDefinition.opacity !== undefined) {
+        maMaterialData.opacity = materialDefinition.opacity;
+    } else {
+        maMaterialData.opacity = DEFAULT_OPACITY;
+    }
+    maOpacity.value = maMaterialData.opacity;
+    maOpacitySlider.value = Math.floor(maMaterialData.opacity * 1000);
+
+    if (materialDefinition.opacityMapMode !== undefined) {
+        maMaterialData.opacityMapMode = materialDefinition.opacityMapMode;
+    } else {
+        maMaterialData.opacityMapMode = "OPACITY_MAP_OPAQUE";
+    }
+    switch(maMaterialData.opacityMapMode) {
```

Comment:
```suggestion
    switch (maMaterialData.opacityMapMode) {
```
And similar elsewhere.

Reply:
Fixed on the 3 switch case.